src/Nethermind/Nethermind.Synchronization/StateSync/PeerInfoExtensions.cs

```diff
@@ -0,0 +1,28 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using Nethermind.Stats.Model;
+using Nethermind.Synchronization.Peers;
+
+namespace Nethermind.Synchronization.StateSync;
+
+public static class PeerInfoExtensions
+{
+    public static bool CanGetNodeData(this PeerInfo peerInfo)
+    {
+        if (!peerInfo.SyncPeer.Node.AgreedCapability.TryGetValue("eth", out int ethVersion))
+        {
+            return false;
+        }
+
+        if (ethVersion <= 66) return true;
+        if (peerInfo.SyncPeer.Node.ClientType == NodeClientType.Nethermind) return true; // Nethermind will still answer to it
+        return false;
+    }
+
+    public static bool CanGetSnapData(this PeerInfo peerInfo)
+    {
+        // Should we like, blacklist Nethermind here?
```

Comment:
I think we could. We are using `snap` protocol only during sync and disabling it just after, so we will not receive anything interesting from nethermind nodes using `snap` protocol because they are not synced

Reply:
Actually we will be disconnected: https://github.com/NethermindEth/nethermind/blob/master/src/Nethermind/Nethermind.Network/P2P/Subprotocols/Snap/SnapProtocolHandler.cs#L166

Reply:
It's a mechanism to not keep peers which are useless for both sides (both not synced)